py/src/braintrust/logger.py

```diff
@@ -1623,6 +1623,16 @@ def _try_log_output(span, output):
 F = TypeVar("F", bound=Callable[..., Any])
 
 
+@overload
```

Comment:
lgtm! @manugoyal mind sanity checking?

Reply:
I mentioned on slack, but a couple of unit tests like will serve as a living record of the contract.  https://github.com/braintrustdata/braintrust-sdk/blob/main/py/src/braintrust/test_logger.py#L596-L632 will serv

Reply:
yah updated tests ... lmk if there is anything else.